lib/storage/metadata/mongoclient/ListRecordStream.js

```diff
@@ -91,23 +91,36 @@ class ListRecordStream extends stream.Readable {
         let entry;
         if (itemObj.op === 'i' &&
             itemObj.o && itemObj.o._id) {
+            const value = itemObj.o.value;
+            if (value.tags) {
```

Comment:
I would think factorizing this code into a separate function `decodeTags` for reuse, at least across the two cases that need it, is worth it.

Reply:
Oh and it can be unit tested more easily then ;)